tomb/src/pipelines/banyan_api/auth.rs

```diff
@@ -15,15 +16,107 @@ pub async fn pipeline(command: AuthSubCommand) -> Result<String> {
     let mut client = global.get_client().await?;
 
     // Process the command
-    let result = match command {
+    let result: Result<String, ClientError> = match command {
+        AuthSubCommand::RegisterDevice => {
+            // let device_key = EcEncryptionKey::generate().await?;
+            let private_device_key = GlobalConfig::from_disk().await?.api_key().await?;
+
+            // Create a public key from the
+            let public_device_key = private_device_key
+                .public_key()
+                .map_err(ClientError::crypto_error)?;
+            // Bytes of the public device key
+            let public_device_key_bytes = public_device_key
+                .export()
+                .await
+                .map_err(ClientError::crypto_error)?;
+
+            // Public device key in PEM format
+            let public_device_key = String::from_utf8(public_device_key_bytes)
+                .expect("cant convert key bytes to string");
+            // Strip the public key of its new lines
+            let mut stripped_public_key = public_device_key.replace('\n', "");
+            // Strip the public key of its prefix and suffix
+            stripped_public_key = stripped_public_key
+                .strip_prefix("-----BEGIN PUBLIC KEY-----")
+                .expect("unable to strip PEM prefix")
+                .strip_suffix("-----END PUBLIC KEY-----")
+                .expect("unable to strip PEM suffix")
+                .to_string();
+
+            // Represent the weird b64 characters with ones that are url-valid
+            let encoded_public_key = stripped_public_key
+                .replace('+', "-")
+                .replace('/', "_")
+                .replace('=', ".")
+                .to_string();
+
+            // Create a new nonce to identify this registration
+            let nonce = Uuid::new_v4();
+            // Create a new request object with the nonce
+            let start_regwait = Regwait { nonce };
+            // Create a base64 url encoded version of the nonce
+            let b64_nonce = base64_url::encode(&nonce);
+            // Create a clone of the client to move into the handle
+            let mut client_1 = client.clone();
+            // Create a join handle for later use, starting the call immediately
+            let join_handle: JoinHandle<Result<RegwaitResponse, ClientError>> =
+                tokio::task::spawn(async move {
+                    // Build the request
+                    client_1.call(start_regwait).await
+                });
+
+            // Should be this in prod TODO
+            // https://alpha.data.banyan.computer/
```

Comment:
Yeah this will need to be configurable somehow. We're likely going to want it as part of the CLI args with a default of whatever our current "production environment" is.